cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -41,6 +41,24 @@ namespace flatbuf = cudf::io::parquet::flatbuf;
 
 namespace {
 
+/**
+ * @brief Computes the total number of row groups in input span of row group indices
+ */
+[[nodiscard]] size_type compute_total_row_groups(
+  host_span<std::vector<size_type> const> row_group_indices)
+{
+  auto const total_row_groups =
+    std::accumulate(row_group_indices.begin(),
+                    row_group_indices.end(),
+                    size_t{0},
+                    [](auto sum, auto const& row_groups) { return sum + row_groups.size(); });
+
+  // Check if we have less than 2B total row groups.
+  CUDF_EXPECTS(total_row_groups <= std::numeric_limits<size_type>::max(),
+               "Total number of row groups exceed the cudf::size_type's limit");
```

Comment:
```suggestion
               "Total number of row groups exceed the column size limit",
               std::overflow_error);
```
This is the standard error message we have been using for this.